cluster-autoscaler/cloudprovider/digitalocean/digitalocean_node_group.go

```diff
@@ -0,0 +1,269 @@
+		return fmt.Errorf("couldn't increase size to %d (delta: %d). Current size is: %d",
+			targetSize, delta, updatedNodePool.Count)
+	}
+
+	return nil
+}
+
+// DeleteNodes deletes nodes from this node group (and also increasing the size
+// of the node group with that). Error is returned either on failure or if the
+// given node doesn't belong to this node group. This function should wait
+// until node group size is updated. Implementation required.
+func (n *NodeGroup) DeleteNodes(nodes []*apiv1.Node) error {
+	ctx := context.Background()
+	for _, node := range nodes {
+		nodeID, ok := node.Labels[nodeIDLabel]
+		if !ok {
+			// CA creates fake node objects to represent upcoming VMs that haven't
+			// registered as nodes yet. They have node.Spec.ProviderID set. Use
+			// that as nodeID.
+			nodeID = node.Spec.ProviderID
+		}
+
+		_, err := n.client.DeleteNode(ctx, n.clusterID, n.id, nodeID, nil)
+		if err != nil {
+			return fmt.Errorf("deleting node failed for cluster: %q node pool: %q node: %q: %s",
+				n.clusterID, n.id, nodeID, err)
+		}
+	}
+
+	return nil
+}
+
+// DecreaseTargetSize decreases the target size of the node group. This function
+// doesn't permit to delete any existing node and can be used only to reduce the
+// request for new nodes that have not been yet fulfilled. Delta should be negative.
+// It is assumed that cloud provider will not delete the existing nodes when there
+// is an option to just decrease the target. Implementation required.
+func (n *NodeGroup) DecreaseTargetSize(delta int) error {
+	if delta >= 0 {
+		return fmt.Errorf("delta must be negative, have: %d", delta)
+	}
+
+	targetSize := n.nodePool.Count + delta
+	if targetSize <= n.MinSize() {
+		return fmt.Errorf("size decrease is too small. current: %d desired: %d min: %d",
+			n.nodePool.Count, targetSize, n.MinSize())
+	}
+
+	req := &godo.KubernetesNodePoolUpdateRequest{
+		Count: targetSize,
+	}
+
+	ctx := context.Background()
+	updatedNodePool, _, err := n.client.UpdateNodePool(ctx, n.clusterID, n.id, req)
+	if err != nil {
+		return err
+	}
+
+	if updatedNodePool.Count != targetSize {
+		return fmt.Errorf("couldn't increase size to %d (delta: %d). Current size is: %d",
+			targetSize, delta, updatedNodePool.Count)
+	}
+
+	return nil
+}
+
+// Id returns an unique identifier of the node group.
+func (n *NodeGroup) Id() string {
+	return n.id
+}
+
+// Debug returns a string containing all information regarding this node group.
+func (n *NodeGroup) Debug() string {
+	return fmt.Sprintf("cluster ID: %s (min:%d max:%d)", n.Id(), n.MinSize(), n.MaxSize())
+}
+
+// Nodes returns a list of all nodes that belong to this node group.  It is
+// required that Instance objects returned by this method have Id field set.
+// Other fields are optional.
+func (n *NodeGroup) Nodes() ([]cloudprovider.Instance, error) {
+	if n.nodePool == nil {
+		return nil, errors.New("node pool instance is not created")
+	}
+	return toInstances(n.nodePool.Nodes), nil
```

Comment:
To get proper error handling you should return 'placeholder' instances for any instance that doesn't exist yet (n.nodePool.Count - len(n.nodePool.Nodes). Unless these is always 0?

It's fine to add it in a separate PR later. This PR is a recent example of this: https://github.com/kubernetes/autoscaler/pull/2235 and could be used as a reference.